### PR TITLE
invite members dialog error message update

### PR DESCRIPTION
**Pull request by karkhypov**

### CODE REVIEW

File: frontend/src/helpers/types/resError.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 export interface resError {
   error_message: string;
-  additional_information?: { [key: string]: any };
+  additional_information?: { errors?: { loc: string[]; msg: string; type: string }[] };
```
Comment (lironhamra) [inline frontend/src/helpers/types/resError.ts:3]: why ? did the err format changed on the server ? 
@yromanyshyn please review it also

File: frontend/src/components/WorkspaceSettings/components/Members/components/InviteMember.tsx [hunk 2]
```diff
@@ -48,18 +48,21 @@ export const InviteMember = ({ open, closeDialog }: MembersActionDialog) => {
 
   const handleInviteMember = async () => {
     const res = await inviteUser({ data: { email: convertEmailsIntoAnArray(email) } });
-    setErr((res as resError)?.error_message ?? 'none');
-  };
 
-  useEffect(() => {
-    if (err === 'none') {
+    if ((res as resError)?.error_message) {
+      const errorMessage = (res as resError).additional_information?.errors?.map(e => e.msg).join(', ');
```
Comment (lironhamra) [inline frontend/src/components/WorkspaceSettings/components/Members/components/InviteMember.tsx:53]: even if the format has changed, it's logic  that should be handled on the server, error_message should includes all the info user needed.

**Review by lironhamra — CHANGES_REQUESTED** [review]

File: frontend/src/helpers/types/resError.ts (hunk 1 above)
Comment (karkhypov) [reply frontend/src/helpers/types/resError.ts:3]: `{
    "error_message": "2 validation errors for Request body -> email str type expected (type=type_error.str) body -> email -> 0 value is not a valid email address (type=value_error.email)",
    "additional_information": {
        "errors": [
            {
                "loc": [
                    "body",
                    "email"
                ],
                "msg": "str type expected",
                "type": "type_error.str"
            },
            {
                "loc": [
                    "body",
                    "email",
                    0
                ],
                "msg": "value is not a valid email address",
                "type": "value_error.email"
            }
        ]
    }
}`

File: frontend/src/components/WorkspaceSettings/components/Members/components/InviteMember.tsx (hunk 2 above)
Comment (karkhypov) [reply frontend/src/components/WorkspaceSettings/components/Members/components/InviteMember.tsx:53]: It includes but not in the way we need it, I have FE ticket for this.
But you are right, this kind of things
Screenshot 2023-06-01 at 14 35 11
 should be handled on the BE.

File: frontend/src/helpers/types/resError.ts (hunk 1 above)
Comment (karkhypov) [reply frontend/src/helpers/types/resError.ts:3]: "errors": [
            {
                "loc": [
                    "body",
                    "email"
                ],
                "msg": "str type expected",
                "type": "type_error.str"
            },
            {
                "loc": [
                    "body",
                    "email",
                    0
                ],
                "msg": "value is not a valid email address",
                "type": "value_error.email"
            }
        ]
    }
}`

Comment (karkhypov) [reply frontend/src/helpers/types/resError.ts:3]: `
{
    "error_message": "2 validation errors for Request body -> email str type expected (type=type_error.str) body -> email -> 0 value is not a valid email address (type=value_error.email)",
    "additional_information": {
        "errors": [
            {
                "loc": [
                    "body",
                    "email"
                ],
                "msg": "str type expected",
                "type": "type_error.str"
            },
            {
                "loc": [
                    "body",
                    "email",
                    0
                ],
                "msg": "value is not a valid email address",
                "type": "value_error.email"
            }
        ]
    }
}
`

Comment (karkhypov) [reply frontend/src/helpers/types/resError.ts:3]: `
{
    "error_message": "2 validation errors for Request body -> email str type expected (type=type_error.str) body -> email -> 0 value is not a valid email address (type=value_error.email)",
    "additional_information": {
        "errors": [
            {
                "loc": [
                    "body",
                    "email"
                ],
                "msg": "str type expected",
                "type": "type_error.str"
            },
            {
                "loc": [
                    "body",
                    "email",
                    0
                ],
                "msg": "value is not a valid email address",
                "type": "value_error.email"
            }
        ]
    }
}
`

Comment (lironhamra) [reply frontend/src/helpers/types/resError.ts:3]: @karkhypov Please never change global type before you are checking all the endpoints/making sure with the server team if its the same scheme always.

Comment (karkhypov) [reply frontend/src/helpers/types/resError.ts:3]: Ok, I tried to find it in the `generated.ts` but looks like there is no such scheme there.

File: frontend/src/components/WorkspaceSettings/components/Members/components/InviteMember.tsx (hunk 2 above)
Comment (lironhamra) [reply frontend/src/components/WorkspaceSettings/components/Members/components/InviteMember.tsx:53]: I commented on the relevant linear issue

File: frontend/src/helpers/types/resError.ts (hunk 1 above)
Comment (lironhamra) [reply frontend/src/helpers/types/resError.ts:3]: @yromanyshyn Your assistence please

**Pull request closed without merging**